tests/ppc_disasm.py

```diff
@@ -0,0 +1,568 @@
+import struct
+import sys
+import capstone as cs
+from typing import Any, BinaryIO, List, Optional, Set, TextIO, Tuple
+from elf_file import ElfFile, ElfSection, ElfSymbol
+
+# Based on `doldisasm.py` from camthesaxman:
+# https://gist.github.com/camthesaxman/a36f610dbf4cc53a874322ef146c4123
+
+
+# Minimal stubs for Capstone
+# fmt: off
+CsMnemonic = int
+CsRegister = int
+class CsInsn:
+    def __init__(self, cs: Any, all_info: Any) -> None: ...
+    @property
+    def id(self) -> CsMnemonic: ...
+    @property
+    def address(self) -> int: ...
+    @property
+    def bytes(self) -> bytes: ...
+    @property
+    def mnemonic(self) -> str: ...
+    @property
+    def op_str(self) -> str: ...
+    @property
+    def operands(self) -> List[Any]: ...
+    def reg_name(self, reg_id: CsRegister, default: Optional[str] = ...) -> str: ...
+# fmt: on
+
+
+MANUAL_MNEMONICS: Set[CsMnemonic] = {
+    # Unsupported instructions
+    cs.ppc.PPC_INS_VMSUMSHM,
+    cs.ppc.PPC_INS_VMHADDSHS,
+    cs.ppc.PPC_INS_XXSLDWI,
+    cs.ppc.PPC_INS_VSEL,
+    cs.ppc.PPC_INS_XVSUBSP,
+    cs.ppc.PPC_INS_XXSEL,
+    cs.ppc.PPC_INS_XVMULSP,
+    cs.ppc.PPC_INS_XVDIVSP,
+    cs.ppc.PPC_INS_VADDUHM,
+    cs.ppc.PPC_INS_XXPERMDI,
+    cs.ppc.PPC_INS_XVMADDASP,
+    cs.ppc.PPC_INS_XVMADDMSP,
+    cs.ppc.PPC_INS_XVCMPGTSP,
+    cs.ppc.PPC_INS_XXMRGHD,
+    cs.ppc.PPC_INS_XSMSUBMDP,
+    cs.ppc.PPC_INS_XSTDIVDP,
+    cs.ppc.PPC_INS_XVADDSP,
+    cs.ppc.PPC_INS_XVCMPEQSP,
+    cs.ppc.PPC_INS_XVMSUBASP,
+    cs.ppc.PPC_INS_XVCMPGESP,
+    # Instructions that Capstone gets wrong
+    cs.ppc.PPC_INS_MFESR,
+    cs.ppc.PPC_INS_MFDEAR,
+    cs.ppc.PPC_INS_MTESR,
+    cs.ppc.PPC_INS_MTDEAR,
+    cs.ppc.PPC_INS_MFICCR,
+    cs.ppc.PPC_INS_MFASR,
+}
+
+LOAD_STORE_MNEMONICS: Set[CsMnemonic] = {
+    cs.ppc.PPC_INS_LWZ,
+    cs.ppc.PPC_INS_LMW,
+    cs.ppc.PPC_INS_LHA,
+    cs.ppc.PPC_INS_LHAU,
+    cs.ppc.PPC_INS_LHZ,
+    cs.ppc.PPC_INS_LHZU,
+    cs.ppc.PPC_INS_LBZ,
+    cs.ppc.PPC_INS_LBZU,
+    cs.ppc.PPC_INS_LFD,
+    cs.ppc.PPC_INS_LFDU,
+    cs.ppc.PPC_INS_LFS,
+    cs.ppc.PPC_INS_LFSU,
+    cs.ppc.PPC_INS_STW,
+    cs.ppc.PPC_INS_STWU,
+    cs.ppc.PPC_INS_STMW,
+    cs.ppc.PPC_INS_STH,
+    cs.ppc.PPC_INS_STHU,
+    cs.ppc.PPC_INS_STB,
+    cs.ppc.PPC_INS_STBU,
+    cs.ppc.PPC_INS_STFS,
+    cs.ppc.PPC_INS_STFSU,
+    cs.ppc.PPC_INS_STFD,
+    cs.ppc.PPC_INS_STDU,
+}
+
+# These name substitutions are performed by doldisasm.py so that all symbols
+# and labels in the output asm only use alphanumerics and '$'
+SUBSTITUTIONS: Tuple[Tuple[str, str], ...] = (
+    ("<", "$$0"),
+    (">", "$$1"),
+    ("@", "$$2"),
+    ("\\", "$$3"),
+    (",", "$$4"),
```

Comment:
(huh, so backslashes and commas can actually occur? I guess we'll actually have to fix the parsing at some point then)

Reply:
I added this for the `@`s. (We could try to support parsing them directly, but it's a bit tricky with `@` being used as a macro sigil)

I know `,`s occur in templates, but overall I figured it was better to match doldisasm.py here.